components/uniset/src/builder.rs

```diff
@@ -43,52 +66,24 @@ impl UnicodeSetBuilder {
             return;
         }
         if self.intervals.is_empty() {
-            self.append(start, end);
+            self.intervals.extend_from_slice(&[start, end]);
             return;
         }
         let last_ind = self.intervals.len() - 1;
```

Comment:
Optional: In `add`, you special-case when the whole interval is before or after the existing table, but you don't have this special case in `remove`.  Consider either adding this special case to `remove` or deleting it from `add`, since I think it is handled by the `.splice()` in `add_remove_middle()`.

Reply:
For now I removed the special case from `add`, going with the philosophy of not optimizing when not proven to be actually "optimal"